docker-compose.yml

```diff
@@ -0,0 +1,28 @@
+version: '3.8'
+
+services:
+  app:
+    build:
+      context: .
+      args:
+        NODE_VERSION: ${NODE_VERSION:-18}
+    ports:
+      - "${PORT:-3000}:3000"
+    environment:
+      - NODE_ENV=development
+    volumes:
+      - .:/app
+      - /app/node_modules
+    command: ["npm", "run", "start:dev"]
+
+  ngrok:
+    image: ngrok/ngrok:latest
+    ports:
+      - "4040:4040"
+    command:
+      - "http"
+      - "app:${PORT:-3000}"
```

Comment:
The application inside the app container always starts on port `3000`. Therefore, there's no need to use the PORT value from the environment variable. Changing the `PORT` value causes Ngrok to throw an error.